python/sglang/srt/layers/vocab_parallel_embedding.py

```diff
@@ -549,6 +549,13 @@ def __init__(
             use_presharded_weights=use_presharded_weights,
         )
         self.quant_config = quant_config
+
+        from sglang.srt.utils import PackWeightMethod
+
+        # We only support pack LMHead if it's not quantized. For LMHead with quant_config, the weight_name will be "qweight"
+        if self.quant_config is None:
```

Comment:
Why is there no if condition for the CPU here?

Reply:
We will check the device inside `PackWeightMethod`. But to make this cleaner, I updated the code to check `_is_cpu and _is_cpu_amx_available` here before setting `self.quant_method`.